vm/http_test.go

```diff
@@ -0,0 +1,66 @@
+package vm
+
+import (
+	"testing"
+	//"net/http/httptest"
+	//"net/http"
+	"net/http"
+	"io/ioutil"
+)
+
+func TestHTTPObject(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		//test get request
+		{`
+		require "net/http"
+
+		Net::HTTP.get("http://127.0.0.1:3000")
+		`, "GET Hello World"},
+		{`
+		require "net/http"
+
+		Net::HTTP.post("http://127.0.0.1:3000", "text/plain", "Hi Again")
+		`, "POST Hi Again"},
+	}
+
+	c := make(chan bool, 1)
+
+	//server to test off of
+	go func() {
+		m := http.NewServeMux()
+
+		m.HandleFunc("/", func (w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+
+			if r.Method == http.MethodPost {
+				b, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					panic(err)
+				}
+				w.Write([]byte(r.Method + " " + string(b)))
+			} else {
+				w.Write([]byte(r.Method + " Hello World"))
```

Comment:
Same as above

Reply:
I think using r.Method on this line is preferable as it can be used to test DELETE, HEAD, etc.

Reply:
I see, I didn't know you will also add those methods.